cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client_test.go

```diff
@@ -110,32 +109,41 @@ func TestWaitForOpError(t *testing.T) {
 	defer server.Close()
 	g := newTestAutoscalingGceClient(t, "project1", server.URL, "")
 
-	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationDoneResponseError).Once()
-
-	operation := &gce_api.Operation{Name: "operation-1505728466148-d16f5197"}
+	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197/wait").Return(operationDoneResponseError).Once()
 
-	err := g.waitForOp(operation, projectId, zoneB, false)
+	err := g.WaitForOperation("operation-1505728466148-d16f5197", "TestWaitForOpError", projectId, zoneB)
 	assert.Error(t, err)
+	mock.AssertExpectationsForObjects(t, server)
 }
 
 func TestWaitForOpTimeout(t *testing.T) {
 	server := test_util.NewHttpServerMock()
 	defer server.Close()
 	g := newTestAutoscalingGceClient(t, "project1", server.URL, "")
 
-	// The values here are higher than in other tests since we're aiming for timeout.
-	// Lower values make this fragile and flakey.
-	g.operationPollInterval = 10 * time.Millisecond
-	g.operationWaitTimeout = 49 * time.Millisecond
-
-	// Sometimes, only 3 calls are made, but it doesn't really matter,
-	// so let's not assert expectations for this mock, just check for timeout error.
-	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationRunningResponse).Times(5)
+	// default polling interval and wait time are too big for the test
+	g.operationWaitTimeout = 1 * time.Millisecond
```

Comment:
With the timeout being this low, the test looks even more identical to `TestWaitForOpContextTimeout` - I think in both we're testing if timing out the API call via context works correctly (since 1ms likely isn't enough to obtain the result, so the call will just fail). Or am I not seeing some difference here?

Reply:
I think you are right that such low timeout values actually make them look the same, I've adjusted those slightly to be more easily differentiated. The main difference between these tests is that `TestWaitForOpContextTimeout` should never achieve http client receiving response (to better reflect that - I've added a minute delay and a comment), TestWaitForOpContextTimeout should receive the response and respond with an error that operation is still running